server/resp/src/main/java/org/infinispan/server/resp/Resp3Handler.java

```diff
@@ -43,313 +21,31 @@ protected void setCache(AdvancedCache<byte[], byte[]> cache) {
-                        removes.incrementAndGet();
-                     }
-                  }));
-         }
-         return stageToReturn(deleteStages.freeze(), ctx, (removals, alloc) -> {
-            stringToByteBuf(":" + removals.get() + "\r\n", alloc);
-         });
-      }
-   }
-
-   private CompletionStage<RespRequestHandler> performMget(ChannelHandlerContext ctx, Cache<byte[], byte[]> cache, List<byte[]> arguments) {
-      int keysToRetrieve = arguments.size();
-      if (keysToRetrieve == 0) {
-         stringToByteBuf("*0\r\n", allocatorToUse);
-         return myStage;
-      }
-      List<byte[]> results = Collections.synchronizedList(Arrays.asList(
-            new byte[keysToRetrieve][]));
-      AtomicInteger resultBytesSize = new AtomicInteger();
-      AggregateCompletionStage<Void> getStage = CompletionStages.aggregateCompletionStage();
-      for (int i = 0; i < keysToRetrieve; ++i) {
-         int innerCount = i;
-         byte[] keyBytes = arguments.get(i);
-         getStage.dependsOn(cache.getAsync(keyBytes)
-               .whenComplete((returnValue, t) -> {
-                  if (returnValue != null) {
-                     results.set(innerCount, returnValue);
-                     int length = returnValue.length;
-                     if (length > 0) {
-                        // $ + digit length (log10 + 1) + /r/n + byte length
-                        resultBytesSize.addAndGet(1 + (int) Math.log10(length) + 1 + 2 + returnValue.length);
-                     } else {
-                        // $0 + /r/n
-                        resultBytesSize.addAndGet(2 + 2);
-                     }
-                  } else {
-                     // $-1
-                     resultBytesSize.addAndGet(3);
-                  }
-                  // /r/n
-                  resultBytesSize.addAndGet(2);
-               }));
-      }
-      return stageToReturn(getStage.freeze(), ctx, (ignore, alloc) -> {
-         int elements = results.size();
-         // * + digit length (log10 + 1) + \r\n + accumulated bytes
-         int byteAmount = 1 + (int) Math.log10(elements) + 1 + 2 + resultBytesSize.get();
-         ByteBuf byteBuf = alloc.apply(byteAmount);
-         byteBuf.writeCharSequence("*" + results.size(), CharsetUtil.US_ASCII);
-         byteBuf.writeByte('\r');
-         byteBuf.writeByte('\n');
-         for (byte[] value : results) {
-            if (value == null) {
-               byteBuf.writeCharSequence("$-1", CharsetUtil.US_ASCII);
-            } else {
-               byteBuf.writeCharSequence("$" + value.length, CharsetUtil.US_ASCII);
-               byteBuf.writeByte('\r');
-               byteBuf.writeByte('\n');
-               byteBuf.writeBytes(value);
-            }
-            byteBuf.writeByte('\r');
-            byteBuf.writeByte('\n');
-         }
-         assert byteBuf.writerIndex() == byteAmount;
-      });
+   public <K, V> AdvancedCache ignorePreviousValuesCache() {
```

Comment:
is generic really needed here? it produces warnings around the code
```suggestion
   public AdvancedCache<byte[], byte[]> ignorePreviousValuesCache() {
```

Reply:
yes